plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/grouped-product-selector/edit.tsx

```diff
@@ -19,21 +24,49 @@ export default function AddToCartWithOptionsGroupedProductSelectorEdit(
 	props: BlockEditProps< Attributes >
 ) {
 	const { className } = props.attributes;
-	const { current: currentProductType } = useProductTypeSelector();
 	const { product } = useProductDataContext();
-	const productType =
-		product.id === 0 ? currentProductType?.slug : product.type;
-
 	const blockProps = useBlockProps( {
 		className,
 	} );
 	const innerBlocksProps = useInnerBlocksProps( blockProps, {
 		template: GROUPED_PRODUCT_ITEM_TEMPLATE,
 		templateLock: 'all',
 	} );
+	const [ groupedProduct, setGroupedProduct ] = useState<
+		ProductResponseItem[]
+	>( [] );
+
+	// If product ID is 0, then we must be editing a template.
+	// Fetch an existing grouped product so template can be edited.
+	useEffect( () => {
+		if ( product.id !== 0 && product.type === 'grouped' ) {
+			setGroupedProduct( [ product as ProductResponseItem ] );
+		}
+		if ( groupedProduct.length === 0 && product.id === 0 ) {
+			resolveSelect( productsStore )
+				.getProducts( {
+					type: 'grouped',
+					per_page: 3,
+				} )
+				.then( ( fetchedProduct ) => {
+					if ( fetchedProduct.length > 0 ) {
+						setGroupedProduct( fetchedProduct );
+					}
+				} );
+		}
+	}, [ groupedProduct, product ] );
 
-	if ( productType !== 'grouped' ) {
-		return null;
+	if ( groupedProduct.length === 0 ) {
+		return (
+			<div { ...innerBlocksProps }>
+				<NoticeBanner status="warning">
+					{ __(
+						'No grouped products were found. Please create a grouped product first.',
+						'woocommerce'
+					) }
+				</NoticeBanner>
+			</div>
+		);
 	}
```

Comment:
If no Grouped Products are found, I think it would be better to fall back to displaying 3 random products in the _Grouped Product Selector_ block, what do you think?

Reply:
I've removed the `NoticeBanner` and instead changed this to fallback to displaying 3 products when a grouped product isn't available: https://github.com/woocommerce/woocommerce/pull/57690/commits/f0c2fed8835fe47c1d4c06736c67d5a8662dab7f